### PR TITLE
[23.0] Fix dataype_change not updating HDCA update_time

### PR DESCRIPTION
**Pull request by mvdbeek**
That update_time is used to find changed items in the history

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: lib/galaxy/webapps/galaxy/services/history_contents.py [hunk 1]
```diff
@@ -1428,8 +1429,10 @@ def _change_datatype(
                 wrapped_task = self._change_item_datatype(dataset_instance, params, trans)
                 if wrapped_task:
                     wrapped_tasks.append(wrapped_task)
+            trans.sa_session.execute
```
Comment (nsoranzo) [inline lib/galaxy/webapps/galaxy/services/history_contents.py]: This doesn't seem right?

Comment (mvdbeek) [reply lib/galaxy/webapps/galaxy/services/history_contents.py]: yep, didn't mean to commit that.
```suggestion
```

**Comment by nsoranzo** [comment]
The test_galaxy_packages is due to https://github.com/sqlalchemy/alembic/issues/1243 .

**Comment by nsoranzo** [comment]
Agreed, I am testing a fix right now, then I'll open the PR.

**Comment by mvdbeek** [comment]
awesome, thank you!

**Pull request merged**